packages/ui/src/types/authMachine.ts

```diff
@@ -28,8 +28,8 @@ interface BaseFormContext {
 }
 
 export interface SignInContext extends BaseFormContext {
-  loginMechanisms: AuthContext['config']['loginMechanisms'];
-  socialProviders: AuthContext['config']['socialProviders'];
+  loginMechanisms: Required<AuthContext>['config']['loginMechanisms'];
+  socialProviders: Required<AuthContext>['config']['socialProviders'];
```

Comment:
Is adding `Required` going to cause any of the frameworks to complain if these props aren't passed into the `authenticator`?

Reply:
I don't think so, because the parent machine have default values for `loginMechanisms` and `socialProviders` even if these props weren't passed by the end user: https://github.com/aws-amplify/amplify-ui/blob/896437385d7f514687d6644235df6ed13e7e9402/packages/ui/src/machines/authenticator/index.ts#L164-L172

Reply:
👍